src/apify_client/clients/base/actor_job_base_client.py

```diff
@@ -1,13 +1,60 @@
-from typing import Optional
+import math
+import time
+from datetime import datetime
+from typing import Dict, Optional
 
+from ..._consts import TERMINAL_ACTOR_JOB_STATUSES, ActorJobStatus
+from ..._errors import ApifyApiError
+from ..._utils import _catch_not_found_or_throw, _parse_date_fields, _pluck_data
 from .resource_client import ResourceClient
 
-# from ..._utils import _parse_date_fields, _pluck_data
+DEFAULT_WAIT_FOR_FINISH = 999999
 
 
 class ActorJobBaseClient(ResourceClient):
     """Base sub-client class for actor runs and actor builds."""
 
-    def _wait_for_finish(self, wait_secs: Optional[int] = None) -> None:
-        # TODO implement when implementing run and build endpoints
-        pass
+    def _wait_for_finish(self, wait_secs: Optional[int] = None) -> Optional[Dict]:
+        started_at = datetime.now()
+        should_repeat = True
+        job: Optional[Dict] = None
+        seconds_elapsed = 0
+
+        while(should_repeat):
+            wait_for_finish = DEFAULT_WAIT_FOR_FINISH
+            if wait_secs is not None:
+                wait_for_finish = wait_secs - seconds_elapsed
+
+            try:
+                response = self.http_client.call(
+                    url=self._url(),
+                    method='GET',
+                    params=self._params(waitForFinish=wait_for_finish),
+                )
+                job = _parse_date_fields(_pluck_data(response.json()))
+
+                seconds_elapsed = math.floor(((datetime.now() - started_at).total_seconds()))
+                if (
+                    ActorJobStatus(job['status']) in TERMINAL_ACTOR_JOB_STATUSES or (wait_secs is not None and seconds_elapsed >= wait_secs)
+                ):
+                    should_repeat = False
+
+                if not should_repeat:
+                    # Early return here so that we avoid the sleep below if not needed
+                    return job
+
+                # It might take some time for database replicas to get up-to-date so sleep a bit before retrying
+                time.sleep(250)
+
+            except ApifyApiError as exc:
```

Comment:
This catch should be before the `time.sleep()`, otherwise if the job is not found it will never sleep and just retry the request immediately.

The point here is that when you do `client.actor('actor_id').call(...)`, it will start the actor, wait for it to finish and then return the run object once it's finished. It can happen, though, that when you start the actor and immediately try to access the run object, the newly created run object might not be propagated to all the DB replicas, and the first API request to get that run object might not find it, so we should wait a bit before trying again to give the DB replicas a chance to sync all the changes.

Perhaps we should also improve the `should_repeat` condition to not wait for the DB replica sync for too long, otherwise this might never return even if the run object really doesn't exist.

Reply:
I moved the `sleep` but I'm not sure what you mean by modifying `should_repeat` condition. thx

Reply:
I meant that right now, when you would do `wait_for_finish` on a non-existing actor job, you would be here catching `not found` errors in a loop, stuck forever, instead of just returning `None` if it really doesn't exist. We should allow for an early exit from this loop if there was enough time for the DB replicas to sync and the actor job is still `None`.
I think maybe 1 second should be enough for the replicas to sync? So if the API still returns that the job does not exist even after that time, we should just return None, as the job really does not exist.